app/UI/app_update.tsx

```diff
@@ -214,23 +216,44 @@ const handle_update_event = async (chan: PutChannel<true>, args: {
         //
         else if (event.type == "ViewSpaceSettings") {
             const relay = pool.getRelay(event.url);
-            if (relay) {
-                app.popOverInputChan.put({
-                    children: (
-                        <SpaceSetting
-                            profileGetter={app.database}
-                            emit={app.eventBus.emit}
-                            getSpaceInformationChan={relay.getRelayInformationStream}
-                            getSpaceMembersChan={relay.getSpaceMembersStream}
-                            spaceUrl={relay.url}
-                            getMemberSet={app.database.getMemberSet}
-                            getProfileByPublicKey={app.database.getProfileByPublicKey}
-                        />
-                    ),
-                });
-            } else {
+            if (!relay) {
                 console.error(event.url, "is not in the pool");
+                continue;
             }
+            const getMemberSetChan: func_GetMemberSetChan = (isRelayed) => () => {
+                const chan = csp.chan<Set<string> | Error>();
+                (async () => {
+                    if (!isRelayed) {
+                        await chan.put(app.database.getMemberSet(relay.url)());
+                        await chan.close();
+                        return;
+                    }
+                    const membersStream = relay.getSpaceMembersStream();
+                    for await (const members of membersStream) {
+                        if (members instanceof Error) {
+                            await chan.put(members);
+                        } else {
+                            const pubkeys = new Set(
+                                members.map((event: SpaceMember) => event.member),
+                            );
+                            await chan.put(pubkeys);
+                        }
+                    }
```

Comment:
`mebersStream` does not put anything in this chan when the result remains unchanged. Therefore, it is impossible to close this inner chan after the outer chan is closed.

There are two ways to solve this problem, but both require modifying the inner chan.
1. Do not judge whether the fetch data has changed in the inner layer chan, put all information as it is. In this way, you can determine whether the outer chan is closed in the outer loop and then close the inner chan.
1. Add `Symbol("No change");` to the inner chan so that the outer loop can also close the inner chan.

Reply:
Just add some code that when the outter channel is closed, close the inner channel as well.